docs/KIP-0030/KIP-0030.md

```diff
@@ -37,398 +37,181 @@ Wallets and dApps will communicate with the end user's browser via `https`. All
-
-```json
-{
-  "capabilities": {
-    "CREATE_ORDER_LINE": {
-      "granter": {
-        "argIndex": 2
-      },
-      "acceptor": {
-        "argIndex": 3
-      },
-      "hashValues": [0, 2, 3, 4],
-      "hashIndex": 1
-    }
-  }
-}
+The Wallet is responsible for connecting existing accounts or help the user to
+create a new account. How the user is guided through this process is up to the
+Wallet.
+
+After the user has connected a new account or selected an existing account, the
+Wallet should emit an `postMessage` event on the `window` of the Wallet. In the
+message the Wallet should include the following information:
+
+```ts
+type ConnectEventPayload = {
+  source: 'kadena-spirekey';
+  name: 'connected';
+  payload: Account;
+};
+```
+
+If the user cancels the connection process, the Wallet should close the window
+and emit an event with the following information:
+
+```ts
+type CanceledConnectEventPayload = {
+  source: 'kadena-spirekey';
+  name: 'canceled:connect';
+  payload: void;
+};
+```
+
+## Sign
+
+The dApp will provide a button to sign a transaction. The SpireKey SDK will
+display a notification another window has activity. This allows the wallet to be
+opened in a new window and and ask the user to sign the transaction. The Wallet
+will be opened on the path `/sign`. The Wallet is responsible for displaying the
+transaction to the user and ask for the user's approval. How the user is guided
+through this process is up to the Wallet.
+
+The dApp will send a list of transactions to the Wallet and optionally a list of
+accounts that need to have fungibles avaible for the transaction. The Wallet can
+use this information to help the user funnel the funds to the requested chain.
+
+The parameters will be send to the Wallet in the form of hash parameters. The
+parameters are: `transactions` and `accounts`. The `transactions` parameter is
+expected to be a JSON encoded array of transactions. The `accounts` parameter is
+expected to be a JSON encoded array of `OptimalTransactionsAccount` which is a
+subset of an `Account`.
+
+After the user has signed for all necessary transactions, the Wallet should
+submit any transactions that are needed to funnel the funds to the requested
+chain. The TransactionDescriptors for those transactions should be included in
+the `postMessage` event.
+
+After the user has signed for all necessary transactions, the Wallet should emit
+an `postMessage` event on the `window` of the Wallet. In the message the Wallet
+should include the following information:
+
+```ts
+type SignEventPayload = {
+  source: 'kadena-spirekey';
+  name: 'signed';
+  payload: {
+    accounts: Account[];
+    txs: Record<string, { sig: string; pubKey: string }[]>;
+  };
+};
+```
+
+If the user cancels the sign process, the Wallet should close the window and
+emit an event with the following information:
+
+```ts
+type CanceledSignEventPayload = {
+  source: 'kadena-spirekey';
+  name: 'canceled:sign';
+  payload: void;
```

Comment:
`void` is not valid in this context. should probably be `null` or `undefined` or the key left out entirely.